test/staBalPaths.spec.ts

```diff
@@ -0,0 +1,355 @@
+// TS_NODE_PROJECT='tsconfig.testing.json' npx mocha -r ts-node/register test/staBalPaths.spec.ts
```

Comment:
Many of these tests are pretty much copy pastes from each other. Can you deduplicate these where you're just passing in different inputs to the same code?

This is something that we need to be doing generally with this codebase. The amount of duplication in the tests is huge.

Reply:
Deduplicated and added more thorough Path testing.